### PR TITLE
Merge display for hint mode to rest of HintMode logic

### PR DESCRIPTION
**Pull request by coreylevine**
Merge the fix for displaying the hint to the main hint mode branch.

### CODE REVIEW

**Comment by svelupillai** [comment]
I checked everything out, it looks good, there is a few lines of commented out code that we do not need. I will delete them and approve the changes.

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java [hunk 1]
```diff
@@ -330,6 +386,14 @@ public void setQuestionFont(String questionFont) {
         this.questionFont = questionFont;
     }
 
+    public String getHintFont() {
+        return answerFont;
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Are we using the same database fiend for the hint and answer font?

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java [hunk 2]
```diff
@@ -330,6 +386,14 @@ public void setQuestionFont(String questionFont) {
...
+    }
+
+    public void setHintFont(String answerFont) {
+        this.answerFont = answerFont;
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Same here

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java (hunk 1 above)
Comment (svelupillai) [reply app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: the same font yes

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java [hunk 3]
```diff
@@ -372,9 +436,10 @@ public void setAnswerAudioLocation(String answerAudioLocation) {
 
     public boolean isDefaultColor() {
         return questionTextColor == null &&
-               answerTextColor == null &&
-               questionBackgroundColor == null &&
-               answerBackgroundColor == null &&
-               separatorColor.equals(DEFAULT_SEPARATOR_COLOR);
+                //hintTextColor == null &&
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Remove commented out code.

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java [hunk 4]
```diff
@@ -372,9 +436,10 @@ public void setAnswerAudioLocation(String answerAudioLocation) {
...
+                answerTextColor == null &&
+                questionBackgroundColor == null &&
+                //hintBackgroundColor == null &&
+                answerBackgroundColor == null && separatorColor.equals(DEFAULT_SEPARATOR_COLOR);
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Remove commented out code.

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java (hunk 3 above)
Comment (svelupillai) [reply app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: will do

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java (hunk 4 above)
Comment (svelupillai) [reply app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: mhm girl

File: app/src/main/java/org/liberty/android/fantastischmemo/ui/StudyActivity.java [hunk 5]
```diff
@@ -419,6 +417,16 @@ protected boolean onClickAnswerText() {
         return true;
     }
 
+    @Override
+    protected boolean onClickHintText() {
+        if (!isHintShown()) {
+            onClickHintView();
+        }else {
```
Comment (coreylevine) [inline app/src/main/java/org/liberty/android/fantastischmemo/ui/StudyActivity.java]: Add a space between } and else

File: app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java (hunk 2 above)
Comment (coreylevine) [reply app/src/main/java/org/liberty/android/fantastischmemo/entity/Setting.java]: Same answer as above

**Comment by j-suguma** [comment]
The CI build past so you can go ahead and merge
image

**Pull request merged**